### PR TITLE
GLSL 330 shaders

### LINKED ISSUE #1721: Mesa 21 without Mesa workaround incorrectly renders PointCloud2
Describe your issue here and explain how to reproduce it.

### Your environment
* OS Version: Ubuntu 18.04
* ROS Distro: Melodic
* RViz, Qt, OGRE, OpenGl version as printed by rviz:
    ```
    [ INFO] [1646652899.525315495] [ros.rviz]: rviz version 1.13.24
    [ INFO] [1646652899.525380866] [ros.rviz]: compiled against Qt version 5.9.5
    [ INFO] [1646652899.525397976] [ros.rviz]: compiled against OGRE version 1.9.0 (Ghadamon)
    [ INFO] [1646652899.533460029] [ros.rviz]: Forcing OpenGl version 0.
    [ INFO] [1646652899.757093342] [ros.rviz]: Stereo is NOT SUPPORTED
    [ INFO] [1646652899.757155850] [ros.rviz]: OpenGL device: AMD RENOIR (DRM 3.44.0, 5.16.10-051610-generic, LLVM 13.0.0)
    [ INFO] [1646652899.757174358] [ros.rviz]: OpenGl version: 4.6 (GLSL 4.6).
    ```
* If source build, which git commit? 0462dd27
* System locale, i.e. the output of `echo "$LANG $LC_NUMERIC"`: `C C`

On this system, I'm using a more recent Mesa from kisak-mesa PPA (mesa 21.3.7).

Currently, there is the mesa version workaround that forces using GLSL 1.4 on systems with Mesa >= 20. In order to resolve #1701, I tried disabling the Mesa workaround, since the original issue (large points) should have been fixed since Mesa 21.0.3. The big points issue is fixed, but there is another problem: every style of PointCloud2 display except "Points" is displayed just as a white dot regardless of the settings:

rviz_screenshot_2022_03_07-12_35_45

### How to test

- run `rviz` in one terminal, configure a `PointCloud2` display for topic `points2`, set fixed frame to `map`.
- in a second terminal, run this script: https://github.com/ros-visualization/rviz/blob/noetic-devel/src/test/send_point_cloud_2.py
- play around with the pointcloud display style

### PR DESCRIPTION
**Pull request by simonschmeisser**
This comes with a lot of hubris and a lack of fundamental knowledge so please be patient and don't hold back anything that comes up when you see this lame attempt ...

Also this is a WIP of course, a proper solution would either try to add 330 besides 150 in proper folders or remove 150 altogether.

I'm trying to fix #1721 and just noticed your parallel attempt @rhaschke 

This was verified with GPUpad syntactically but I could not get neither gpupad nor rviz to render anything (or not crash in the case of rviz ...)

@paroj may I ask you for help here or in #1725

### CODE REVIEW

**Comment by simonschmeisser** [comment]
I was taking some information from https://ogrecave.github.io/ogre/api/latest/_high-level-_programs.html

File: ogre_media/materials/glsl150/box.geom [hunk 1]
```diff
@@ -19,13 +19,13 @@ uniform mat4 worldviewproj_matrix;
 uniform vec4 size;
 uniform vec4 auto_size;
 
-in gl_PerVertex {
-	vec4 gl_Position;
-	vec4 gl_FrontColor;
-} gl_in[];
+in PerVertex {
+	vec4 frontColor;
+} gs_in[];
```
Comment (paroj) [inline ogre_media/materials/glsl150/box.geom:24]: I think these are the only strictly necessary changes in this PR

**Comment by rhaschke** [comment]
@simonschmeisser, GPUpad seems to be an interesting tool to develop the GLSL scripts.
However, I couldn't find an option to save a whole session (like the examples). Did you?

**Comment by simonschmeisser** [comment]
Simply File > Save as   It mostly references external files though. So you first need to save all of the scripts and then the project

**Comment by simonschmeisser** [comment]
image
Using 1.23.1 Appimage

**Comment by rhaschke** [comment]
Hm. I'm able to save a newly created session. But I cannot save (as) an example session?

**Comment by rhaschke** [comment]
Closing in favor of #1725, which seems to work with Mesa (and a hack in OGRE).

**Pull request closed without merging**